jax/_src/numpy/lax_numpy.py

```diff
@@ -3712,35 +3712,47 @@ def argpartition(a: ArrayLike, kth: int, axis: int = -1) -> Array:
 
 
 @partial(jit, static_argnums=(2,))
-def _roll(a, shift, axis):
-  a_shape = shape(a)
-  if axis is None:
-    return lax.reshape(_roll(ravel(a), shift, axis=0), a_shape)
-  shift = asarray(shift)
-  a_ndim = len(a_shape)
-  axis = np.asarray(axis)
-  b_shape = lax.broadcast_shapes(shift.shape, axis.shape, (1,))
+def _roll(a: Array, shift: Array, axis: Sequence[int]) -> Array:
+  b_shape = lax.broadcast_shapes(shift.shape, np.shape(axis))
   if len(b_shape) != 1:
     msg = "'shift' and 'axis' arguments to roll must be scalars or 1D arrays"
     raise ValueError(msg)
 
   for x, i in zip(broadcast_to(shift, b_shape),
                   np.broadcast_to(axis, b_shape)):
-    i = _canonicalize_axis(i, a_ndim)
-    a_shape_i = array(a_shape[i], dtype=np.int32)
+    a_shape_i = array(a.shape[i], dtype=np.int32)
     x = ufuncs.remainder(lax.convert_element_type(x, np.int32),
-                  lax.max(a_shape_i, np.int32(1)))
-    a = lax.concatenate((a, a), i)
-    a = lax.dynamic_slice_in_dim(a, a_shape_i - x, a_shape[i], axis=i)
+                         lax.max(a_shape_i, np.int32(1)))
+    a_concat = lax.concatenate((a, a), i)
+    a = lax.dynamic_slice_in_dim(a_concat, a_shape_i - x, a.shape[i], axis=i)
   return a
 
+@partial(jit, static_argnums=(1, 2))
+def _roll_static(a: Array, shift: Sequence[int], axis: Sequence[int]) -> Array:
+  for ax, s in zip(*np.broadcast_arrays(axis, shift)):
```

Comment:
Either you don't need `np.broadcast_arrays` here or the type annotation is wrong. I'm guessing more likely the later?

Reply:
It's correct as written. `shift` and `axis` are sequences, but they may not have the same shape, and `np.broadcast_arrays` accepts sequences as arguments in place of arrays.